docs/developers/data/abi.md

```diff
@@ -279,3 +279,576 @@ Similarly, [enums](/developers/data/custom-types#custom-enums) are defined by:
+    fn add(&self, value: BigUint) {
+        self.sum().update(|sum| *sum += value);
+    }
+}
+```
+
+Adding the `#[esdt_attribute("testBasic", BigUint)]` at trait level along with `#[multiversx_sc::contract]` should export a new structure named `testBasic` with a `BigUint` field type. The abi can be generated calling `sc-meta all abi` in the contract folder, or by building the contract using `sc-meta all build` (this command also adds the wasm file to the `output` folder).
+
+Building the contract using `sc-meta all build` will generate the following folder structure:
+```
+adder
+├── output
+│   ├── adder.abi.json
+│   ├── adder.imports.json
+|   ├── adder.mxsc.json
+|   ├── adder.wasm
+│   ├── testBasic.esdt-abi.json
+```
+
+Let's check out the `adder.abi.json` file first. Here we discover the new `esdtAttributes` field, containing the value mentioned in the annotation.
+
+```json
+{
+    "buildInfo": {
+        "rustc": {
+            "version": "1.71.0-nightly",
+            "commitHash": "a2b1646c597329d0a25efa3889b66650f65de1de",
+            "commitDate": "2023-05-25",
+            "channel": "Nightly",
+            "short": "rustc 1.71.0-nightly (a2b1646c5 2023-05-25)"
+        },
+        "contractCrate": {
+            "name": "adder",
+            "version": "0.0.0",
+            "gitVersion": "v0.44.0"
+        },
+        "framework": {
+            "name": "multiversx-sc",
+            "version": "0.44.0"
+        }
+    },
+    "docs": [
+        "One of the simplest smart contracts possible,",
+        "it holds a single variable in storage, which anyone can increment."
+    ],
+    "name": "Adder",
+    "constructor": {
+        "inputs": [
+            {
+                "name": "initial_value",
+                "type": "BigUint"
+            }
+        ],
+        "outputs": []
+    },
+    "endpoints": [
+        {
+            "name": "getSum",
+            "mutability": "readonly",
+            "inputs": [],
+            "outputs": [
+                {
+                    "type": "BigUint"
+                }
+            ]
+        },
+        {
+            "docs": [
+                "Add desired amount to the storage variable."
+            ],
+            "name": "add",
+            "mutability": "mutable",
+            "inputs": [
+                {
+                    "name": "value",
+                    "type": "BigUint"
+                }
+            ],
+            "outputs": []
+        }
+    ],
+    "events": [],
+    "esdtAttributes": [
+        {
+            "ticker": "testBasic",
+            "type": "BigUint"
+        }
+    ],
+    "hasCallback": false,
+    "types": {}
+}
+```
+We can also check the specific json file exported for the newly defined type where we can find information about the type separated from the main abi file.
+
+__testBasic.esdt-abi.json__
+```json
+{
+    "esdtAttribute": {
+        "ticker": "testBasic",
+        "type": "BigUint"
+    }
+}
+```
+
+## Composite types
```

Comment:
`Using more complex types`

Just give the examples and keep them minimal.

Reply:
Remember, all types are explained in the section above, this is just about the ESDT syntax.